src/main/java/org/opentripplanner/routing/edgetype/TemporaryPartialStreetEdge.java

```diff
@@ -10,140 +10,26 @@
 import org.opentripplanner.routing.vertextype.TemporaryVertex;
 import org.opentripplanner.util.I18NString;
 
-import java.util.List;
-
-
-final public class TemporaryPartialStreetEdge extends StreetWithElevationEdge implements TemporaryEdge {
-
-    private static final long serialVersionUID = 1L;
-
-    /**
-     * The edge on which this lies.
-     */
-    private StreetEdge parentEdge;
-
-
-    /**
-     * Create a new partial street edge along the given 'parentEdge' from 'v1' to 'v2'.
-     * If the length is negative, a new length is calculated from the geometry.
-     * The elevation data is calculated using the 'parentEdge' and given 'length'.
-     */
-    public TemporaryPartialStreetEdge(StreetEdge parentEdge, StreetVertex v1, StreetVertex v2,
-            LineString geometry, I18NString name, double length) {
-        super(v1, v2, geometry, name, length, parentEdge.getPermission(), false);
-        this.parentEdge = parentEdge;
-        setCarSpeed(parentEdge.getCarSpeed());
-        setElevationProfileUsingParents();
-
-        // Assert that the edge is going in the right direction [only possible if vertex is temporary]
-        assertEdgeIsNotDirectedAwayFromTemporaryEndVertex(v1);
-        assertEdgeIsDirectedTowardsTemporaryEndVertex(v2);
-    }
-
-    /**
-     * Create a new partial street edge along the given 'parentEdge' from 'v1' to 'v2'.
-     * The length is calculated using the provided geometry.
-     * The elevation data is calculated using the 'parentEdge' and the calculated 'length'.
-     */
-    TemporaryPartialStreetEdge(StreetEdge parentEdge, StreetVertex v1, StreetVertex v2,
-            LineString geometry, I18NString name) {
-        super(v1, v2, geometry, name, 0, parentEdge.getPermission(), false);
-        this.parentEdge = parentEdge;
-        setCarSpeed(parentEdge.getCarSpeed());
-
-        // No length is known, so we use the provided geometry to estimate it
-        calculateLengthFromGeometry();
-        setElevationProfileUsingParents();
+/**
+ * This class models a StreetEdge that was non-destructively split from another StreetEdge for the purposes of modeling
+ * StreetEdges that should only be valid for a single request.
```

Comment:
I think including an example here would be helpful in distinguishing this further from SemiPermanentPartialStreetEdge. For example, is this used to create an edge from origin to connect to the street network?

Reply:
Yeah, that's their primary use, but I'm not sure if the GTFS-flex code uses this edge type too. I'll add a little comment.